src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -2056,43 +2056,81 @@ public static void townSetBoard(CommandSender sender, String[] split, Town town,
 		TownyMessaging.sendTownBoard(player, town);
 	}
 
+	@Deprecated
 	public static void townSetTitle(CommandSender sender, String[] split, boolean admin, Town town, Resident resident, Player player) throws TownyException {
+		townSetTitle(sender, split, admin);
+	}
+
+	public static void townSetTitle(@NotNull CommandSender sender, @NotNull String[] split, boolean admin) throws TownyException {
 		// Give the resident a title
 		if (split.length < 2)
-			throw new TownyException("Eg: /town set title bilbo Jester ");
+			throw new TownyException("Eg: /town set title bilbo Jester");
 
-		resident = getResidentOrThrow(split[1]);
+		Resident resident = getResidentOrThrow(split[1]);
+		String title = StringMgmt.join(NameValidation.checkAndFilterArray(StringMgmt.remArgs(split, 2)));
+		
+		townSetTitle(sender, resident, title, admin);
+	}
 
-		if (!admin && !CombatUtil.isSameTown(getResidentOrThrow(player.getUniqueId()), resident))
+	/**
+	 * @param sender The command sender who initiated the command.
+	 * @param resident The resident to set the title for.
+	 * @param title The title to set.
+	 * @param admin Whether to skip the same-town test.
+	 * @throws TownyException If the title wasn't able to be set.
+	 */
+	public static void townSetTitle(@NotNull CommandSender sender, @NotNull Resident resident, @NotNull String title, boolean admin) throws TownyException {
+		if (!admin && sender instanceof Player player && !CombatUtil.isSameTown(getResidentOrThrow(player), resident))
 			throw new TownyException(Translatable.of("msg_err_not_same_town", resident.getName()));
-
-		String title = StringMgmt.join(NameValidation.checkAndFilterArray(StringMgmt.remArgs(split, 2)));
+		
+		title = NameValidation.filterName(title);
+		
 		if (title.length() > TownySettings.getMaxTitleLength())
 			throw new TownyException(Translatable.of("msg_err_input_too_long"));
-
+		
 		if (NameValidation.isConfigBlacklistedName(title))
 			throw new TownyException(Translatable.of("msg_invalid_name"));
 
 		resident.setTitle(title);
 		resident.save();
 
 		if (resident.hasTitle())
-			TownyMessaging.sendPrefixedTownMessage(town, Translatable.of("msg_set_title", resident.getName(), Colors.translateColorCodes(resident.getTitle())));
+			TownyMessaging.sendPrefixedTownMessage(resident.getTownOrNull(), Translatable.of("msg_set_title", resident.getName(), Colors.translateColorCodes(resident.getTitle())));

```

Comment:
Only thing I've spotted is that (I think) it would be possible for an admin to run `/ta town NAME set title ....` on a townless player and resident.getTownOrNull() will throw an exception.